### PR TITLE
zh-trans: update static-pod.md

### PR DESCRIPTION
**Pull request by mysunshine92**
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> Remember to delete this note before submitting your pull request.
>
> For pull requests on 1.15 Features: set Milestone to 1.15 and Base Branch to dev-1.15
> 
> For pull requests on Chinese localization, set Base Branch to release-1.14
>
> For pull requests on Korean Localization: set Base Branch to dev-1.14-ko.\<latest team milestone>
>
> If you need Help on editing and submitting pull requests, visit:
> https://kubernetes.io/docs/contribute/start/#improve-existing-content.
>
> If you need Help on choosing which branch to use, visit:
> https://kubernetes.io/docs/contribute/start#choose-which-git-branch-to-use.
>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>

### CODE REVIEW

**Comment by mysunshine92** [comment]
cc @tengqm

**Comment by jsafrane** [comment]
How did I become reviewer of static pods?
Anyway, the English part looks ok, I can't comment on the Chinese part.
/approve

File: content/zh/docs/tasks/administer-cluster/static-pod.md [hunk 1]
```diff
@@ -1,24 +1,74 @@
...
+*Static pods* are managed directly by kubelet daemon on a specific node, without the API server observing it. It does not have an associated replication controller, and kubelet daemon itself watches it and restarts it when it crashes. There is no health check. Static pods are always bound to one kubelet daemon and always run on the same node with it.
+--->
+*静态 pods* 直接由特定节点上的 kubelet 进程来管理，不通过主控节点上的 API 服务
+器。静态 pod 不关联任何 replication controller，它由 kubelet 进程自己来监控，当 pod 崩溃时重启该 pod。对于静态 pod 没有健康检查。静态 pod 始终绑定在某一个 kubelet，并且始终运行在同一个节点上。
+
+<!--
+Kubelet automatically tries to create a *mirror pod* on the Kubernetes API server for each static pod.
+This means that the pods are visible on the API server but cannot be controlled from there.
+--->
+Kubelet 自动为每一个静态 pod 在 Kubernetes 的 API 服务器上创建一个镜像 Pod(*mirror pod*)，因此可以在 API 服务器查询到该 pod，但是不被 API 服务器控制(例如不能删除)。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/static-pod.md]:
```suggestion
Kubelet 自动为每一个静态 pod 在 Kubernetes 的 API 服务器上创建一个*镜像 Pod* (*mirror pod*)，因此可以在 API 服务器查询到该 pod，但是不被 API 服务器控制(例如不能删除)。
```

File: content/zh/docs/tasks/administer-cluster/static-pod.md [hunk 2]
```diff
@@ -1,24 +1,74 @@
...
+<!--
+### Configuration files
+
+The configuration files are just standard pod definitions in json or yaml format in a specific directory. Use `kubelet --pod-manifest-path=<the directory>` to start kubelet daemon or add the `staticPodPath: <the directory>` field in the [KubeletConfiguration file](/docs/tasks/administer-cluster/kubelet-config-file), which periodically scans the directory and creates/deletes static pods as yaml/json files appear/disappear there.
+Note that kubelet will ignore files starting with dots when scanning the specified directory.
+--->
 ### 配置文件
 
-配置文件就是放在特定目录下的标准的 JSON 或 YAML 格式的 pod 定义文件。用`kubelet --pod-manifest-path=<the directory>`来启动 kubelet 进程，kubelet 将会周期扫描<the directory>这个目录，根据这个目录下出现或消失的 YAML/JSON 文件来创建或删除静态 pod。
+配置文件就是放在特定目录下的标准的 JSON 或 YAML 格式的 pod 定义文件。用`kubelet --pod-manifest-path=<the directory>`来启动 kubelet 进程或者在[KubeletConfiguration 文件](/docs/tasks/administer-cluster/kubelet-config-file)中添加`staticPodPath: <the directory>`字段，kubelet 将会周期扫描<the directory>这个目录，根据这个目录下出现或消失的 YAML/JSON 文件来创建或删除静态 pod。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/static-pod.md]:
```suggestion
配置文件就是放在特定目录下的标准的 JSON 或 YAML 格式的 pod 定义文件。用 `kubelet --pod-manifest-path=<the directory>` 来启动 kubelet 进程或者在 [KubeletConfiguration 文件](/docs/tasks/administer-cluster/kubelet-config-file)中添加 `staticPodPath: <the directory>` 字段，kubelet 将会周期扫描 `<the directory>` 这个目录，根据这个目录下出现或消失的 YAML/JSON 文件来创建或删除静态 pod。
```

File: content/zh/docs/tasks/administer-cluster/static-pod.md [hunk 3]
```diff
@@ -1,24 +1,74 @@
...
+    ```
+    [joe@host ~] $ ssh my-node1
+    ```
 
-下面例子用静态 pod 的方式启动一个 nginx 的 Web 服务器：
+2. Choose a directory, say `/etc/kubelet.d` and place a web server pod definition there, e.g. `/etc/kubelet.d/static-web.yaml`:
+--->
+下面例子用静态 pod 的方式启动一个简单的 Web 服务器：
 
 1. 选择一个节点来运行静态 pod。这个例子中就是`my-node1`。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/static-pod.md]:
```suggestion
1. 选择一个节点来运行静态 pod。这个例子中就是 `my-node1`。
```

File: content/zh/docs/tasks/administer-cluster/static-pod.md [hunk 4]
```diff
@@ -1,24 +1,74 @@
...
+1. Choose a node where we want to run the static pod. In this example, it's `my-node1`.
+
+    ```
+    [joe@host ~] $ ssh my-node1
+    ```
 
-下面例子用静态 pod 的方式启动一个 nginx 的 Web 服务器：
+2. Choose a directory, say `/etc/kubelet.d` and place a web server pod definition there, e.g. `/etc/kubelet.d/static-web.yaml`:
+--->
+下面例子用静态 pod 的方式启动一个简单的 Web 服务器：
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/static-pod.md]:
```suggestion
下面例子用静态 pod 的方式启动一个简单的 web 服务器：
```

File: content/zh/docs/tasks/administer-cluster/static-pod.md [hunk 5]
```diff
@@ -28,54 +78,80 @@ Kubelet 自动为每一个静态 pod 在 Kubernetes 的 API 服务器上创建
...
+
+    ```
+    KUBELET_ARGS="--cluster-dns=10.254.0.10 --cluster-domain=kube.local --pod-manifest-path=/etc/kubelet.d/"
     ```
 
-3.配置节点上的 kubelet 使用这个目录，kubelet 启动时增加`--pod-manifest-path=/etc/kubelet.d/`参数。如果是 Fedora 系统，在 Kubelet 配置文件 /etc/kubernetes/kubelet 中添加下面这行：
-	
+    Instructions for other distributions or Kubernetes installations may vary.
+--->
+3.配置节点上的 kubelet 使用这个目录，kubelet 启动时增加`--pod-manifest-path=/etc/kubelet.d/`参数或者在[KubeletConfiguration 文件](/docs/tasks/administer-cluster/kubelet-config-file)中增加`staticPodPath: <the directory>`字段。如果是 Fedora 系统，在 Kubelet 配置文件 /etc/kubernetes/kubelet 中添加下面这行：
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/static-pod.md]:
```suggestion
3.配置节点上的 kubelet 使用这个目录，kubelet 启动时增加`--pod-manifest-path=/etc/kubelet.d/`参数或者在 [KubeletConfiguration 文件](/docs/tasks/administer-cluster/kubelet-config-file)中增加 `staticPodPath: <the directory>` 字段。如果是 Fedora 系统，在 Kubelet 配置文件 /etc/kubernetes/kubelet 中添加下面这行：
```

File: content/zh/docs/tasks/administer-cluster/static-pod.md [hunk 6]
```diff
@@ -28,54 +78,80 @@ Kubelet 自动为每一个静态 pod 在 Kubernetes 的 API 服务器上创建
...
-    [root@my-node1 ~] $ cat <<EOF >/etc/kubelet.d/static-web.yaml
-    apiVersion: v1
-    kind: Pod
-    metadata:
-      name: static-web
-      labels:
-        role: myrole
-    spec:
-      containers:
+```shell
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/static-pod.md]: 这里的代码可以不用修改。

也和其他地方的代码风格一致。:smile:

File: content/zh/docs/tasks/administer-cluster/static-pod.md [hunk 7]
```diff
@@ -28,54 +78,80 @@ Kubelet 自动为每一个静态 pod 在 Kubernetes 的 API 服务器上创建
...
 ## 通过 HTTP 创建静态 Pods
 
 Kubelet 周期地从 --manifest-url=<URL> 参数指定的地址下载文件，并且把它翻译成 JSON/YAML 格式的 pod 定义。此后的操作方式与 --pod-manifest-path=<directory> 相同，kubelet 会不时地重新下载该文件，当文件变化时对应地终止或启动静态 pod(如下)。
 
+<!--
+## Behavior of static pods
+
+When kubelet starts, it automatically starts all pods defined in directory specified in `--pod-manifest-path=` or `--manifest-url=` arguments or add the `staticPodPath: <the directory>` field in the [KubeletConfiguration file](/docs/tasks/administer-cluster/kubelet-config-file), i.e. our static-web.  (It may take some time to pull nginx image, be patient…):
+--->
 ## 静态 pods 的动作行为
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/static-pod.md]:
```suggestion
## 静态 pods 的行为
```

Comment (chenrui333) [reply content/zh/docs/tasks/administer-cluster/static-pod.md]: 另外的改进意见，虽然你没有修改这部分 😆

File: content/zh/docs/tasks/administer-cluster/static-pod.md [hunk 8]
```diff
@@ -85,18 +161,34 @@ NAME                       READY     STATUS    RESTARTS   AGE
...
 需要注意的是，我们不能通过 API 服务器来删除静态 pod(例如，通过 [`kubectl`](/docs/user-guide/kubectl/) 命令)，kebelet 不会删除它。
 
+<!--
+{{< note >}}
+Make sure the kubelet has permission to create the mirror pod in the API server. If not, the creation request is rejected by the API server. See
+[PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/).
+{{< /note >}}
+--->
+{{< note >}}
+需要确保 kubelet 有权限在 API 服务器上创建镜像 pod。如果没有权限，API 服务器会拒绝创建请求，请参考[PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/)。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/static-pod.md]:
```suggestion
需要确保 kubelet 有权限在 API 服务器上创建镜像 pod。如果没有权限，API 服务器会拒绝创建请求，请参考 [PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/)。
```

File: content/zh/docs/tasks/administer-cluster/static-pod.md [hunk 9]
```diff
@@ -85,18 +161,34 @@ NAME                       READY     STATUS    RESTARTS   AGE
...
+pod "static-web-my-node1" deleted
 [joe@my-master ~] $ kubectl get pods
 NAME                       READY     STATUS    RESTARTS   AGE
 static-web-my-node1        1/1       Running   0          12s
 ```
-
+<!--
+Back to our `my-node1` host, we can try to stop the container manually and see, that kubelet automatically restarts it in a while:
+--->
 返回`my-node1`主机，我们尝试手动终止容器，可以看到 kubelet 很快就会自动重启容器。
```
Comment (chenrui333) [inline content/zh/docs/tasks/administer-cluster/static-pod.md]:
```suggestion
返回 `my-node1` 主机，我们尝试手动终止容器，可以看到 kubelet 很快就会自动重启容器。
```

**Comment by chenrui333** [comment]
Thanks for your contributions!

**Comment by mysunshine92** [comment]
@chenrui333 thanks for your suggesstion, I have modified those.
please add an lgtm and approve label, thanks.

**Comment by chenrui333** [comment]
Thanks for the updates! Cheers!

**Pull request merged**